link/src/main/java/com/stripe/android/link/LinkActivityViewModel.kt

```diff
@@ -27,9 +39,23 @@ internal class LinkActivityViewModel : ViewModel() {
         dismissWithResult = null
     }
 
-    internal class Factory : ViewModelProvider.Factory {
-        override fun <T : ViewModel> create(modelClass: Class<T>): T {
-            return LinkActivityViewModel() as T
+    companion object {
+        val Factory: ViewModelProvider.Factory = viewModelFactory {
+            initializer {
+                val savedStateHandle: SavedStateHandle = createSavedStateHandle()
+                val app = this[APPLICATION_KEY] as Application
+                val args: NativeLinkArgs =
+                    requireNotNull(getArgs(savedStateHandle))
```

Comment:
how can we ensure that the args aren't null here when the savedStateHandle was just created? Where are the args set?

Reply:
The `createSavedStateHandle` will obtain arguments from the viewModelStoreOwner, which is `LinkActivity` in this case. The args were set [here](https://github.com/stripe/stripe-android/blob/c58d6e9ff806013eeb49faef8b2bef7f088bd974/link/src/main/java/com/stripe/android/link/LinkActivity.kt#L104). The `LinkActivityContractTest` will make sure these args are always set.

Reply:
gotcha, thank you!

Reply:
FWIW, we shouldn't do these kinds of things. Merchants get mad at us when they see crashes when pen testing tools launch these.

We should add a test that the activity is finished without throwing in the case we don't get the arguments we expect.